### ISSUE TITLE
mongodbatlas_maintenance_window fails with BadRequest: Invalid Day of Week

### ISSUE THREAD

**Issue by languitar** [mongodb/terraform-provider-mongodbatlas#289]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v0.12.27
+ provider.mongodbatlas v0.6.3
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_maintenance_window" "maintenance" {
  project_id  = data.mongodbatlas_project.this.id
  day_of_week = 2
  hour_of_day = 4
}
```

### Steps to Reproduce

I have remodeled the existing configuration of one of our Atlas clusters (shared tier, M5). One aspect of this is the maintenance window. For that, I am using the configuration shown above. Interestingly, the initial plan resulted in this diff (which I don't understand):
```
  # mongodbatlas_maintenance_window.maintenance will be updated in-place
  ~ resource "mongodbatlas_maintenance_window" "maintenance" {
        day_of_week         = 2
        hour_of_day         = 4
        id                  = "<theid>"
        number_of_deferrals = 0
      + project_id          = "<theid>"
        start_asap          = false
    }
```

### Expected Behavior

No diff is shown because the actual settings match.

### Actual Behavior

Applying the above plan results in the following error:

> Error: error updating the MongoDB Atlas Maintenance Window (someid): PATCH https://cloud.mongodb.com/api/atlas/v1.0/groups/theid/maintenanceWindow: 400 (request "Bad Request") Invalid Day of Week. Value should be between 1 and 7 (Sunday = 1).

### Debug Output 

Unfortunately, I can't reproduce this now, as a new iteration of planing doesn't yield a diff anymore.

**Comment by themantissa** [comment]
@languitar thank you for what you have here, helpful, but can you provide more information?  I'm not sure what you mean by remodeled?  Are you trying to change an existing configuration?  For example was maintenance windows a resource you had before and now you want to update it but you are getting an error?  

Also note, while the maintenance window is set at the project level it won't impact shared tier clusters, only M10 or above.

**Comment by languitar** [comment]
We had an existing configuration in Atlas with exactly the settings as described in the HCL file. A terraform configuration didn't exist yet. Therefore,  I imported the resource and afterwards issued a plan and apply cycle.

> Also note, while the maintenance window is set at the project level it won't impact shared tier clusters, only M10 or above.

Good to know. But not a real issue so far. Using the terraform-based configuration here is mostly just a test for applying this also for the production cluster.

**Comment by themantissa** [comment]
Just to double check,  you had an existing project with a maintenance window set and at least an M5, you then imported the 3 items: project, maintenance window and M5.  Then did a plan/apply with the tf config (which matched what was imported) and got this error?  We need to reproduce this hence why the specifics are so important.

**Comment by languitar** [comment]
Yes, that's what I have done.

**Comment by themantissa** [comment]
@languitar thank you!  I was able to reproduce the issue and produce a debug log.  I'll provide the log directly to our developers but the behavior was exact: 1) after import the terraform plan showed the change 2) terraform apply failed and when I ran terraform apply again, with no changes to the tf, it succeeded which is quite odd.   I'll set this as a non-blocking bug for now.

FYI, on the fields that are show - copied below: id to project_id is likely part of the problem.  The other two fields are one we did not configure for Terraform management as they are transient states depending on the system: number_of_deferrals and start_asap.  When a maintenance is available that will honor the desired schedule one can either defer it up to two times or start it asap.  However these really don't fit the terraform philopsy so while returned by the API GET we do not surface them.

        id                  = "<theid>"
        number_of_deferrals = 0
      + project_id          = "<theid>"
        start_asap          = false
    }

**Comment by languitar** [comment]
Good to see that this could be replicated. Did you also get the Atlas Error for the wrong parameter for the day of the week? I can't see that in your diff.

**Comment by themantissa** [comment]
I did, it looked like a pretty spot on replication.

**Comment by themantissa** [comment]
Fixed and included in 0.7.0, closing issue. Thanks!
